diff --git a/app/main.c b/app/main.c
index ff8d684f72..f7531b4261 100644
--- a/app/main.c
+++ b/app/main.c
@@ -401,7 +401,7 @@ gimp_macos_setenv (const char * progname)
       g_free (tmp);
       if (g_getenv ("HOME") != NULL)
         {
-          tmp = g_strdup_printf ("%s/Library/Application Support/GIMP/3.00/cache",
+          tmp = g_strdup_printf ("%s/Library/Caches/dehesselle.GIMP",
                                  g_getenv ("HOME"));
           g_setenv ("XDG_CACHE_HOME", tmp, TRUE);
           g_free (tmp);
diff --git a/libgimpbase/gimpenv.c b/libgimpbase/gimpenv.c
index 6a2ef9a8a5..c588cb57d5 100644
--- a/libgimpbase/gimpenv.c
+++ b/libgimpbase/gimpenv.c
@@ -282,7 +282,7 @@ gimp_directory (void)
       library_dir = [path objectAtIndex:0];
 
       gimp_dir = g_build_filename ([library_dir UTF8String],
-                                   GIMPDIR, GIMP_USER_VERSION, NULL);
+                                   "dehesselle.GIMP", NULL);
 
       [pool drain];
 
